added edit image

Davut97 opened the pull request:
i added option to upload a new image and fixed some bugs

louisrli commented on src/Facebook.js:
```diff
@@ -1,82 +1,138 @@
+  const [isEditing, setIsEditing] = React.useState(false);
+  const [image, setImage] = React.useState(null);
+  const [url, setUrl] = React.useState('');
+  const [progress, setProgress] = React.useState(0);
```
name this variable `uploadProgress` (progress is ambiguous)

louisrli commented on src/Facebook.js:
```diff
@@ -1,82 +1,138 @@
+const EditProfile = (props) => {
+  const [isEditing, setIsEditing] = React.useState(false);
+  const [image, setImage] = React.useState(null);
+  const [url, setUrl] = React.useState('');
```
what is this the url of? better variable name

louisrli commented on src/Facebook.js:
```diff
@@ -1,82 +1,138 @@
     setIsEditing(true);
-  }
+  };
+  const handleChange = (e) => {
```
better variable name, is this handling the change of the inputs or what is it handling the change of?

louisrli commented on src/Facebook.js:
```diff
@@ -1,82 +1,138 @@
+    }
+  };
+  const handleUpload = () => {
+    const uploadTask = storage.ref(`images/${image.name}`).put(image);
```
hm actually i realized i configured the firebase rules wrong. i don't think this works at the moment with the production database, does it?

```

service firebase.storage {
  match /b/{bucket}/o {
    // Files look like: "user/<UID>/path/to/file.txt"
    match /profilePictures/{userId}/{allPaths=**} {
      allow read, write: if request.auth != null && request.auth.uid == userId;
    }
  }
}```

^ see what the expected bucket and such is

Davut97 commented:
I fixed the variables names, and fixed a bug when user clicking upload without an choosing an image the website was crashing, also fixed a bug where if the user clicked submit without entering any info, the city and profile field will be empty.
